core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -451,23 +459,52 @@ private VirtualGroupRequest getVirtualGroupRequest(String virtualGroupsEnvironme
-            sssdConnfig.put("realm", kerberosConfig.getRealm().toUpperCase());
-            sssdConnfig.put("domain", kerberosConfig.getDomain());
-            sssdConnfig.put("password", kerberosConfig.getPassword());
-            sssdConnfig.put("server", kerberosConfig.getUrl());
-            sssdConnfig.put("dns_ttl", kerberosDetailService.getDnsTtl());
+            Map<String, Object> sssdConfig = new HashMap<>();
+            sssdConfig.put("principal", kerberosConfig.getPrincipal());
+            sssdConfig.put("realm", kerberosConfig.getRealm().toUpperCase());
+            sssdConfig.put("domain", kerberosConfig.getDomain());
+            sssdConfig.put("password", kerberosConfig.getPassword());
+            sssdConfig.put("server", kerberosConfig.getUrl());
+            sssdConfig.put("dns_ttl", kerberosDetailService.getDnsTtl());
             // enumeration has performance impacts so it's only enabled if Ranger is installed on the cluster
             // otherwise the usersync does not work with nss
             boolean enumerate = !CollectionUtils.isEmpty(serviceLocations.get("RANGER_ADMIN"))
                     || !CollectionUtils.isEmpty(serviceLocations.get("NIFI_REGISTRY_SERVER"))
                     || !CollectionUtils.isEmpty(serviceLocations.get("NIFI_NODE"));
-            sssdConnfig.put("enumerate", enumerate);
-            servicePillar.put("sssd-ipa", new SaltPillarProperties("/sssd/ipa.sls", singletonMap("sssd-ipa", sssdConnfig)));
+            sssdConfig.put("enumerate", enumerate);
+            Map<String, Object> freeIpaConfig = createFreeIpaConfig(environmentCrn);
+            return Map.of("sssd-ipa", new SaltPillarProperties("/sssd/ipa.sls",
+                    Map.of("sssd-ipa", sssdConfig, "freeipa", freeIpaConfig)));
+        } else {
+            return Map.of();
+        }
+    }
+
+    private Map<String, Object> createFreeIpaConfig(String environmentCrn) {
+        Optional<DescribeFreeIpaResponse> freeIpaResponse = freeipaClient.findByEnvironmentCrn(environmentCrn);
+        if (freeIpaResponse.isPresent()) {
+            Stream<InstanceMetaDataResponse> instanceMetaDataStream = freeIpaResponse.get().getInstanceGroups().stream()
+                    .flatMap(ig -> ig.getMetaData().stream());
+            InstanceMetaDataResponse instanceMetaDataResponse = instanceMetaDataStream
+                    .filter(im -> InstanceStatus.CREATED == im.getInstanceStatus())
+                    .min(Comparator.comparing(InstanceMetaDataResponse::getDiscoveryFQDN))
+                    .orElseGet(() -> instanceMetaDataStream
+                            .min(Comparator.comparing(InstanceMetaDataResponse::getDiscoveryFQDN))
+                            .orElse(null));
+            LOGGER.info("Chosen instance for default FreeIPA server: {}", instanceMetaDataResponse);
+            if (instanceMetaDataResponse == null) {
+                LOGGER.debug("No FreeIPA instance available");
```

Comment:
[nit] maybe this should be "No healthy FreeIPA instance available".

Reply:
as we include unhealthy ones if we can't find a healthy one this would be not true :)